### PR TITLE
Added new tests

### PR DESCRIPTION
**Pull request by ethangraham2001**
- `toMap()` for itinerary that isn't actively used now since we are mocking our repository, but will be used as soon as we definitively integrate with firebase
- OverviewScreen and LikedScreen tests ( will need to add some more for that sweet 80% )
- Changed some test tags for consistency between tests and classes

### CODE REVIEW

File: app/src/test/java/com/github/wanderwise_inc/app/model/location/ItineraryTest.kt
Comment (276746) [inline app/src/test/java/com/github/wanderwise_inc/app/model/location/ItineraryTest.kt:1]: I just wrote test for this class to cover almost everything. I will accepts and add my tests in next PR

**Review by 276746 — APPROVED** [review]

**Review by gunnm0 — COMMENTED** [review]
nice tests, how much coverage did you gain per file approximately? Will need to expand on them from now on, nice work

File: app/src/main/java/com/github/wanderwise_inc/app/ui/itinerary/ItineraryBanner.kt [hunk 1]
```diff
@@ -70,7 +75,7 @@ fun ItineraryBanner(
       elevation = CardDefaults.cardElevation(defaultElevation = 6.dp),
       shape = RoundedCornerShape(13.dp),
       onClick = { onBannerClick() },
-      modifier = Modifier.testTag("Itinerary banner")) {
+      modifier = Modifier.testTag("${ItineraryBannerTestTags.ITINERARY_BANNER}_${itinerary.uid}")) {
```
Comment (ThomasPFMuller) [inline app/src/main/java/com/github/wanderwise_inc/app/ui/itinerary/ItineraryBanner.kt:78]: nice test tag name

File: app/src/test/java/com/github/wanderwise_inc/app/ui/LikedScreenTest.kt [hunk 2]
```diff
@@ -17,73 +17,92 @@ import com.github.wanderwise_inc.app.model.location.Itinerary
...
   }
+
+  @Test
+  fun `a liked itinerary should be displayed`() {
+    composeTestRule
+        .onNodeWithTag("${ItineraryBannerTestTags.ITINERARY_BANNER}_${itinerary.uid}")
+        .isDisplayed()
+  }
 }
- */
```
Comment (ThomasPFMuller) [inline app/src/test/java/com/github/wanderwise_inc/app/ui/LikedScreenTest.kt:89]: Nice simple skeleton, will add more tests in a later PR

File: app/src/test/java/com/github/wanderwise_inc/app/ui/OverviewScreenTest.kt [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+    composeTestRule.onNodeWithTag(OverviewScreenTestTags.CATEGORY_SELECTOR).assertIsDisplayed()
+  }
+
+  @Test
+  fun `at least one itinerary should be displayed`() {
+    composeTestRule
+        .onNodeWithTag("${ItineraryBannerTestTags.ITINERARY_BANNER}_${itinerary.uid}")
+        .isDisplayed()
+  }
+}
```
Comment (ThomasPFMuller) [inline app/src/test/java/com/github/wanderwise_inc/app/ui/OverviewScreenTest.kt:106]: Nice simple skeleton, will add more tests in a later PR

**Review by ThomasPFMuller — APPROVED** [review]
Nice tests, I will probably create a copy in the androidTest folder to create instrumented tests for these screens

**Comment by ethangraham2001** [comment]
> nice tests, how much coverage did you gain per file approximately? Will need to expand on them from now on, nice work

@gunnm0 coverage is a little under 80%. I believe 77%

File: app/src/main/java/com/github/wanderwise_inc/app/ui/itinerary/ItineraryBanner.kt
Comment (engjellismaili) [inline app/src/main/java/com/github/wanderwise_inc/app/ui/itinerary/ItineraryBanner.kt:1]: good idea to create some test tag

File: app/src/test/java/com/github/wanderwise_inc/app/model/location/ItineraryTest.kt
Comment (engjellismaili) [inline app/src/test/java/com/github/wanderwise_inc/app/model/location/ItineraryTest.kt:1]: same comment as before, good idea for test tags

File: app/src/test/java/com/github/wanderwise_inc/app/ui/LikedScreenTest.kt
Comment (engjellismaili) [inline app/src/test/java/com/github/wanderwise_inc/app/ui/LikedScreenTest.kt:1]: Good tests and good use of Mockito, well done

File: app/src/test/java/com/github/wanderwise_inc/app/ui/OverviewScreenTest.kt
Comment (engjellismaili) [inline app/src/test/java/com/github/wanderwise_inc/app/ui/OverviewScreenTest.kt:1]: Good start of test, need to add some test for later

**Review by engjellismaili — COMMENTED** [review]
Overall good practice of testing using mocktio, well done

File: app/src/main/java/com/github/wanderwise_inc/app/ui/list_itineraries/OverviewScreen.kt [hunk 4]
```diff
@@ -22,6 +22,10 @@ import com.github.wanderwise_inc.app.ui.home.SearchBar
 import com.github.wanderwise_inc.app.viewmodel.MapViewModel
 import com.github.wanderwise_inc.app.viewmodel.ProfileViewModel
 
+object OverviewScreenTestTags {
+  const val CATEGORY_SELECTOR = "category selector"
```
Comment (YofiY) [inline app/src/main/java/com/github/wanderwise_inc/app/ui/list_itineraries/OverviewScreen.kt:26]: lgtm !

File: app/src/test/java/com/github/wanderwise_inc/app/model/location/ItineraryTest.kt [hunk 5]
```diff
@@ -4,56 +4,33 @@ import org.junit.Assert.*
...
-  @Test operator fun component8() {}
-
-  @Test fun copy() {}
-
-  @Test fun testToString() {}
-
-  @Test fun testHashCode() {}
-
-  @Test fun testEquals() {}
+  private val dummyItinerary =
```
Comment (YofiY) [inline app/src/test/java/com/github/wanderwise_inc/app/model/location/ItineraryTest.kt:7]: good basic cases with 0 values

File: app/src/test/java/com/github/wanderwise_inc/app/model/location/ItineraryTest.kt [hunk 6]
```diff
@@ -4,56 +4,33 @@ import org.junit.Assert.*
...
+            "title" to "test",
+            "tags" to listOf(ItineraryTags.ADVENTURE),
+            "description" to "test",
+            "visible" to false,
+            "price" to 0f,
+            "time" to 0,
+        )
+
+    assertEquals(expected, actual)
+  }
```
Comment (YofiY) [inline app/src/test/java/com/github/wanderwise_inc/app/model/location/ItineraryTest.kt:35]: Excellent job Ethan !

File: app/src/test/java/com/github/wanderwise_inc/app/ui/LikedScreenTest.kt [hunk 7]
```diff
@@ -17,73 +18,103 @@ import com.github.wanderwise_inc.app.model.location.Itinerary
...
       profileViewModel = ProfileViewModel(profileRepository, imageRepository)
 
-      LikedScreen(mapViewModel, profileViewModel)
+      navHostController = mock(NavHostController::class.java)
+      firebaseAuth = mock(FirebaseAuth::class.java)
+      firebaseUser = mock(FirebaseUser::class.java)
+      `when`(firebaseAuth.currentUser).thenReturn(firebaseUser)
+      `when`(firebaseUser.uid).thenReturn(null)
+
+      LikedScreen(mapViewModel, profileViewModel, navHostController, firebaseAuth)
```
Comment (YofiY) [inline app/src/test/java/com/github/wanderwise_inc/app/ui/LikedScreenTest.kt:98]: looks very good to me

**Review by YofiY — APPROVED** [review]
Overall excellent tests, nothing to add.

**Pull request merged**